internal/component/database_observability/postgres/collector/query_tables_test.go

```diff
@@ -0,0 +1,520 @@
+				"some_database",
+			}},
+			logsLabels: []model.LabelSet{
+				{"job": database_observability.JobName, "op": OP_QUERY_ASSOCIATION, "instance": "postgres-db"},
+				{"job": database_observability.JobName, "op": OP_QUERY_PARSED_TABLE_NAME, "instance": "postgres-db"},
+				{"job": database_observability.JobName, "op": OP_QUERY_PARSED_TABLE_NAME, "instance": "postgres-db"},
+				{"job": database_observability.JobName, "op": OP_QUERY_PARSED_TABLE_NAME, "instance": "postgres-db"},
+			},
+			logsLines: []string{
+				"level=\"info\" queryID=\"abc123\" query_text=\"SELECT * FROM (SELECT id, name FROM employees_us_east UNION SELECT id, name FROM employees_us_west) AS employees_us UNION SELECT id, name FROM employees_emea\" datName=\"some_database\" engine=\"postgres\"",
+				`level="info" queryID="abc123" datName="some_database" table="employees_us_east" engine="postgres"`,
+				`level="info" queryID="abc123" datName="some_database" table="employees_us_west" engine="postgres"`,
+				`level="info" queryID="abc123" datName="some_database" table="employees_emea" engine="postgres"`,
+			},
+		},
+		{
+			name: "show create table",
+			eventStatementsRows: [][]driver.Value{{
+				"abc123",
+				"SHOW CREATE TABLE some_table",
+				"some_database",
+			}},
+			logsLabels: []model.LabelSet{
+				{"job": database_observability.JobName, "op": OP_QUERY_ASSOCIATION, "instance": "postgres-db"},
+				{"job": database_observability.JobName, "op": OP_QUERY_PARSED_TABLE_NAME, "instance": "postgres-db"},
+			},
+			logsLines: []string{
+				"level=\"info\" queryID=\"abc123\" query_text=\"SHOW CREATE TABLE some_table\" datName=\"some_database\" engine=\"postgres\"",
+				`level="info" queryID="abc123" datName="some_database" table="some_table" engine="postgres"`,
+			},
+		},
+		{
+			name: "show variables",
+			eventStatementsRows: [][]driver.Value{{
+				"abc123",
+				"SHOW VARIABLES LIKE $1",
+				"some_database",
+			}},
+			logsLabels: []model.LabelSet{
+				{"job": database_observability.JobName, "op": OP_QUERY_ASSOCIATION, "instance": "postgres-db"},
+			},
+			logsLines: []string{
+				"level=\"info\" queryID=\"abc123\" query_text=\"SHOW VARIABLES LIKE $1\" datName=\"some_database\" engine=\"postgres\"",
+			},
+		},
+		{
+			name: "query is truncated",
+			eventStatementsRows: [][]driver.Value{{
+				"abc123",
+				"SELECT * FROM some_table WHERE",
+				"some_database",
+			}},
+			logsLabels: []model.LabelSet{
+				{"job": database_observability.JobName, "op": OP_QUERY_ASSOCIATION, "instance": "postgres-db"},
+				{"job": database_observability.JobName, "op": OP_QUERY_PARSED_TABLE_NAME, "instance": "postgres-db"},
+			},
+			logsLines: []string{
+				"level=\"info\" queryID=\"abc123\" query_text=\"SELECT * FROM some_table WHERE\" datName=\"some_database\" engine=\"postgres\"",
+				`level="info" queryID="abc123" datName="some_database" table="some_table" engine="postgres"`,
+			},
```

Comment:
Can we add some WITH ... SELECT, WITH ... DELETE, WITH ... INSERT statements for completeness?